dfc.providerportal.findanapprenticeship/src/dfc.providerportal.findanapprenticeship/Functions/GetApprenticeshipsAsProvider.cs

```diff
@@ -37,23 +37,21 @@ public GetApprenticeshipsAsProvider(IAppCache cache, IApprenticeshipService appr
         [FunctionName("GetApprenticeshipsAsProvider")]
         public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "bulk/providers")]HttpRequest req, ILogger log)
         {
-            List<Apprenticeship> persisted = null;
-
             try
             {
-                log.LogInformation($"[{DateTime.UtcNow:G}] Retrieving Apprenticeships...");
-                
-                persisted = (List<Apprenticeship>)await _apprenticeshipService.GetLiveApprenticeships();
-                
-                if (persisted == null)
-                {
-                    return new EmptyResult();
-                }
-
                 var result = await _cache.GetOrAddAsync<(string ExportKey, string Value)>("DasProviders", async () =>
                 {
+                    log.LogInformation($"Retrieving Apprenticeships...");
+
+                    var apprenticeships = (List<Apprenticeship>)await _apprenticeshipService.GetLiveApprenticeships();
+
+                    if (apprenticeships == null)
+                    {
+                        throw new NullReferenceException($"{nameof(apprenticeships)} cannot be null.");
```

Comment:
```suggestion
                        throw new Exception($"{nameof(apprenticeships)} IApprenticeshipService.GetLiveApprenticeships returned null.");
```
I only expect null ref exceptions to happen when someone hasn't null-proofed property/method access, this could would lead me to an incorrect assumption if I was troubleshooting. I suggest explaining the actual problem in the message as suggested.

Reply:
If you don't want to go for `Exception` then maybe add a custom `DependencyFailure` exception class or something, unless you can think of a more appropriate standard exception to use. Personally I wouldn't bother with a custom one unless there's any expectation of catching it explicitly.

Reply:
Agreed.